Add `no-unnecessary-polyfills` rule (#233)
sindresorhus/eslint-plugin-unicorn: docs/rules/no-unnecessary-polyfills.md

# Enforce the use of built-in methods instead of unnecessary polyfills

This rules helps to use existing methods instead of using extra polyfills. 

## Fail

package.json
```json
{
	"engines": {
		"node": ">=8"
	}
}

```
```js
const objectAssign = require('object-assign');
```

## Pass

package.json
```json
{
	"engines": {

> 		"node": "<4.0.0"

Reviewer: Nobody would use a semver range like this here.

	}
}

```
```js

> const assign = require("object-assign"); // passes as Object.assign is not supported

Reviewer: Single-quotes.

```

## Options

By default it takes node version from `package.json`. However you can override
target node version:

```js

> "unicorn/no-unnecessary-polyfills": ["error", {"targetVersion": "6"}]

Reviewer: I think the option should be called `nodeVersion` and you need to document that it can be a semver range. The example here is also wrong as `6` will only match exactly Node.js `6.0.0`, not `6.0.1`.

> # Enforce to use built in method instead of unnecessary polyfills

Reviewer: ```suggestion
# Enforce the use of built-in methods instead of unnecessary polyfills
```

Takes node version from `package.json` and warns to use supported built in methods

> instead of using unnecessary polyfills.

Reviewer: This could be written better.

> The desired name is configurable, but defaults to `error`.

Reviewer: ?

## Fail

package.json

> ```json5

Reviewer: ```suggestion
```json
```

{
	"engines": {

> 		"node": ">8.0.0"

Reviewer: ```suggestion
		"node": ">=8"
```

	}
}

```
```js

> const assign = require("object-assign"); // warns to use built in Object.assign

Reviewer: ```suggestion
const objectAssign = require('object-assign');
```
